port/unix/omrsignal.c

```diff
@@ -1587,19 +1587,21 @@ initializeSignalTools(OMRPortLibrary *portLibrary)
 
 #if !defined(J9ZOS390)
 #if defined(OSX)
-	/* OSX only has named semaphores. They are not shared across processes, so unlink immediately. */
-	portLibrary->str_printf(portLibrary, semNames[0], 128, "/omr/wakeUpASyncReporter-%d", getpid());
-#define SIGSEM_NAME(_i) semNames[_i]
+	/* OSX only has named semaphores. They are not shared across processes, so unlink immediately.
+	 * The semaphore name length must fit within the SEM_NAME_LEN (31) limit.
+	 */
+	portLibrary->str_printf(portLibrary, semName, sizeof(semName), "/omr/wUASRptr%x-%x", (uint32_t)getpid(), (uint32_t)portLibrary->time_nano_time(portLibrary));
```

Comment:
I don't know if this implies the creation of a directory (/omr) internally: maybe there's a race doing that. These semaphores aren't visible in the file-system so `"/omr-wUASRptr%x-%x"` might be a better pattern in that it doesn't present the opportunity for such a race condition.

Reply:
Works for me, I've made that change.

Reply:
Started some testing via https://openj9-jenkins.osuosl.org/job/Pipeline-Build-Test-Personal/69/

Reply:
Pushing another change caused the build to fail, trying again.
https://openj9-jenkins.osuosl.org/job/Pipeline-Build-Test-Personal/70/

Reply:
The OpenJ9 sanity.functional testing passed.